src/network/networkd-network.c

```diff
@@ -170,8 +171,34 @@ int network_verify(Network *network) {
                 network->routes_by_section = hashmap_free_with_destructor(network->routes_by_section, route_free);
         }
 
-        if (network->link_local < 0)
-                network->link_local = network->bridge ? ADDRESS_FAMILY_NO : ADDRESS_FAMILY_IPV6;
+        if (network->link_local < 0) {
+                network->link_local = ADDRESS_FAMILY_IPV6;
+
+                if (network->bridge)
+                        network->link_local = ADDRESS_FAMILY_NO;
+                else {
+                        NetDev *netdev;
+
+                        HASHMAP_FOREACH(netdev, network->stacked_netdevs) {
+                                MacVlan *m;
+
+                                if (netdev->kind == NETDEV_KIND_MACVLAN)
+                                        m = MACVLAN(netdev);
+                                else if (netdev->kind == NETDEV_KIND_MACVTAP)
+                                        m = MACVTAP(netdev);
+                                else
+                                        // Can passthru MACVLAN/MACVTAP be "co-stacked" with other kinds of netdev?
+                                        break;
+
+                                if (m->mode == NETDEV_MACVLAN_MODE_PASSTHRU)
+                                        network->link_local = ADDRESS_FAMILY_NO;
+                                else
+                                        /* There won't be a passthru MACVLAN/MACVTAP if there's already one in another mode */
+                                        break;
```

Comment:
If this comment is true, then `else` is not necessary. But, I think, in this case, also continuing the loop is safer.

Reply:
https://github.com/torvalds/linux/blob/v5.13/drivers/net/macvlan.c#L1451

I'd like to keep this one, even when the performance impact might be trivial.